### PR TITLE
Log session name when opening/closing session

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Log session name when opening/closing session
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/Session.scala [hunk 1]
```diff
@@ -32,6 +32,7 @@ trait Session {
   def user: String
   def password: String
   def ipAddress: String
+  def name: String
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/Session.scala:35]: mv to right behind handle

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala [hunk 2]
```diff
@@ -139,7 +139,9 @@ class KyuubiSyncThriftClient private (
       .map(_.get("kyuubi.engine.id"))
 
     engineAliveProbeClient.foreach { aliveProbeClient =>
+      val sessionName = SessionHandle.apply(_remoteSessionHandle).identifier + "_aliveness_probe"
```
Comment (yaooqinn) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:142]: what is this for？

Comment (turboFei) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:142]: to bound with the main thrift session.

**Comment by yaooqinn** [comment]
cc @wForget

File: kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala (hunk 2 above)
Comment (yaooqinn) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/client/KyuubiSyncThriftClient.scala:142]: ok

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 3]
```diff
@@ -121,7 +121,7 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
     if (session == null) {
       throw KyuubiSQLException(s"Invalid $sessionHandle")
     }
-    info(s"$sessionHandle is closed, current opening sessions $getOpenSessionCount")
+    info(s"$sessionHandle/${session.name} is closed, current opening sessions $getOpenSessionCount")
```
Comment (wForget) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:124]: ditto

File: kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala [hunk 4]
```diff
@@ -100,7 +100,7 @@ abstract class SessionManager(name: String) extends CompositeService(name) {
       val handle = session.handle
       session.open()
       setSession(handle, session)
-      info(s"$user's session with $handle is opened, current opening sessions" +
+      info(s"$user's session with $handle/${session.name} is opened, current opening sessions" +
```
Comment (wForget) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/session/SessionManager.scala:103]: nit: session.name is not always configured, which may make the log dirty.

**Review by wForget — APPROVED** [review]
LGTM, thanks

**Comment by turboFei** [comment]
thanks, merged to master

**Pull request closed without merging**